clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -63,7 +68,9 @@ public void add(NetworkClient.InFlightRequest request) {
      * Get the oldest request (the one that will be completed next) for the given node
      */
     public NetworkClient.InFlightRequest completeNext(String node) {
-        return requestQueue(node).pollLast();
+        NetworkClient.InFlightRequest inFlightRequest = requestQueue(node).pollLast();
```

Comment:
Although unlikely, but maybe it is safer to only decrement only when pollLast() did not return null.

Reply:
requestQueue(node) will throw an exception if the associated deque does not exist or is empty in length.  So unit testing the place with nulls can not happen, however, I have added a unit test to check that an exception is thrown when accessing nodes that InFlightRequests does not know anything about.

Reply:
Good point. In that case, it seems we don't need to do the null check here. Sorry for the back and forth.